src/main/java/cdw/springtraining/gatekeeper/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,47 @@
+package cdw.springtraining.gatekeeper.security;
+
+
+import cdw.springtraining.gatekeeper.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import cdw.springtraining.gatekeeper.entites.User;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+
+    UserRepository userRepository;
+    @Autowired
+    public CustomUserDetailsService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    @Transactional
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+
+        User user = (User) userRepository.findByUserName(username)
+                .orElseThrow(() -> new UsernameNotFoundException("User not exists by Username or Email"));
+
+        Set<GrantedAuthority> authorities = user.getRoles().stream()
+                .map((role) -> new SimpleGrantedAuthority(role.getRoleName()))
+                .collect(Collectors.toSet());
+
+        return new org.springframework.security.core.userdetails.User(
```

Comment:
is there any reason for using by this way of implementation?

Reply:
Everytime a user is created, Their respective role is added to the list. By using this implementation, access the roles list becomes easier